### DISCUSSION TITLE
chunk对象添加前缀存储到object storage

### DISCUSSION THREAD

**Discussion by gaodq** [juicedata/juicefs#554] [Ideas]
为每个 file 添加一个 tag 属性存放到attr中，这个 file 的每个 chunk 将 tag 作为对象存储 key 的 prefix
比如 file1 的 tag 是 "debug"，那么它的对象 key 可以为 "chunks/debug/0/0/1_0_4"
这样的话可以根据对象存储不同的 prefix 做更细的计费规则
这功能会考虑添加吗？

**Comment by xiaogaozi** [comment]
建议把这个功能做在 JuiceFS 之外，在存储路径里增加 tag 的话会给系统增加不必要的复杂度。

**Comment by davies** [comment]
chunk 的 key 不应该受 文件的 tag 影响，否则 当文件的 tag 变化时需要改变对象存储的key, 将会非常慢。

请问你这个 tag 是怎么来的呢？看能否用独立的 volume 来实现？

**Reply by gaodq** [reply]
用户 id 当做 tag，这个需求是业务特需的，想了一下juicefs没必要添加这个功能，谢谢回复
